### PR TITLE
[BUG] Fix field error in information_schema.columns

### LINKED ISSUE #4857: Field show error in information_schema.columns
Reference from https://dev.mysql.com/doc/refman/8.0/en/information-schema-columns-table.html
CHARACTER_MAXIMUM_LENGTH and CHARACTER_OCTET_LENGTH is error
IS_NULLABLE field is always "NO"

### PR DESCRIPTION
**Pull request by HangyuanLiu**
FIX #4857 
## Proposed changes

Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request. If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce to Doris?
_Put an `x` in the boxes that apply_

- [] Bugfix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation Update (if none of the other choices apply)
- [] Code refactor (Modify the code structure, format the code, etc...)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [] I have create an issue on (Fix #ISSUE), and have described the bug/feature there in detail
- [] Compiling and unit tests pass locally with my changes
- [] I have added tests that prove my fix is effective or that my feature works
- [] If this change need a document change, I have updated the document
- [] Any dependent changes have been merged

## Further comments

If this is a relatively large or complex change, kick off the discussion at dev@doris.apache.org by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: be/src/exec/schema_scanner/schema_columns_scanner.cpp [hunk 1]
```diff
@@ -250,20 +265,33 @@ Status SchemaColumnsScanner::fill_one_row(Tuple *tuple, MemPool *pool) {
...
     // CHARACTER_OCTET_LENGTH
+    // For string columns, the maximum length in bytes.
     {
         int data_type = _desc_result.columns[_column_index].columnDesc.columnType;
         if (data_type == TPrimitiveType::VARCHAR || data_type == TPrimitiveType::CHAR) {
             void *slot = tuple->get_slot(_tuple_desc->slots()[9]->tuple_offset());
             int64_t* str_slot = reinterpret_cast<int64_t*>(slot);
             if (_desc_result.columns[_column_index].columnDesc.__isset.columnLength) {
-                *str_slot = _desc_result.columns[_column_index].columnDesc.columnLength;
+                *str_slot = _desc_result.columns[_column_index].columnDesc.columnLength * 4;
```
Comment (morningman) [inline be/src/exec/schema_scanner/schema_columns_scanner.cpp:291]: Why 4?

File: be/src/exec/schema_scanner/schema_columns_scanner.cpp [hunk 2]
```diff
@@ -236,9 +238,22 @@ Status SchemaColumnsScanner::fill_one_row(Tuple *tuple, MemPool *pool) {
     {
         void *slot = tuple->get_slot(_tuple_desc->slots()[6]->tuple_offset());
         StringValue* str_slot = reinterpret_cast<StringValue*>(slot);
-        str_slot->len = strlen("NO") + 1;
-        str_slot->ptr = (char *)pool->allocate(str_slot->len);
-        memcpy(str_slot->ptr, "NO", str_slot->len);
+
+        if (_desc_result.columns[_column_index].columnDesc.__isset.isAllowNull) {
+            if (_desc_result.columns[_column_index].columnDesc.isAllowNull) {
+                str_slot->len = strlen("YES");
```
Comment (morningman) [inline be/src/exec/schema_scanner/schema_columns_scanner.cpp:244]: No need to plus 1?

File: be/src/exec/schema_scanner/schema_columns_scanner.cpp (hunk 1 above)
Comment (HangyuanLiu) [reply be/src/exec/schema_scanner/schema_columns_scanner.cpp:291]: For mysql-compatible purposes, we need to multiply this by 4. 
According to https://dev.mysql.com/doc/refman/8.0/en/information-schema-columns-table.html. CHARACTER_OCTET_LENGTH stands for maximum length in **bytes**, while CHARACTER_MAXIMUM_LENGTH stands for maximum length in **characters**

File: be/src/exec/schema_scanner/schema_columns_scanner.cpp (hunk 2 above)
Comment (HangyuanLiu) [reply be/src/exec/schema_scanner/schema_columns_scanner.cpp:244]: I don't think +1 is necessary here, and +1 will cause the character matching of the SQL result to fail. Such as IS_NULLABLE = "YES" is false

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**